addons/auth_totp_mail_enforce/controllers/home.py

```diff
@@ -0,0 +1,28 @@
+from datetime import datetime, timedelta
+
+import odoo.addons.auth_totp.controllers.home
+
+from odoo import http
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+
+
+class Home(odoo.addons.auth_totp.controllers.home.Home):
+    @http.route()
+    def web_totp(self, redirect=None, **kwargs):
+        response = super().web_totp(redirect=redirect, **kwargs)
+        if not response.is_qweb or response.qcontext['user'].totp_enabled:
+            return response
```

Comment:
I'd be happy to see an explicit `assert request.session.pre_uid and not request.session.uid` below here, just in case the logic of `super()` evolves.

Also maybe a comment to explain `response.is_qweb` (i.e. when the response is a qweb render and not a pure redirect). Or perhaps we could check  `response.status_code != 200` instead?